File: python/1021.py
n = float(input())
b = int(n * 100)

n100 = b // 10000
b = b%10000
n50 = b // 5000
b = b%5000
n20 = b // 2000
b = b%2000
n10 = b // 1000
b = b%1000
n5 = b // 500
b = b%500
n2 = b // 200
b = b%200
m100 = b // 100
b = b%100
m50 = b // 50
b = b%50
m25 = b // 25
b = b%25
m10 = b // 10
b = b%10
m5 = b // 5
b = b%5


print(f"NOTAS:\n{n100} nota(s) de R$ 100.00\n{n50} nota(s) de R$ 50.00\n{n20} nota(s) de R$ 20.00\n{n10} nota(s) de R$ 10.00\n{n5} nota(s) de R$ 5.00\n{n2} nota(s) de R$ 2.00\nMOEDAS:\n{m100} moeda(s) de R$ 1.00\n{m50} moeda(s) de R$ 0.50\n{m25} moeda(s) de R$ 0.25\n{m10} moeda(s) de R$ 0.10\n{m5} moeda(s) de R$ 0.05\n{b} moeda(s) de R$ 0.01")

File: dart/1021.dart
import 'dart:io';

void main() {
  double n = double.parse(stdin.readLineSync());
  n = n * 100;

  print("NOTAS:");
  print("${n ~/ 10000} nota(s) de R\$ 100.00");
  n = n % 10000;
  print("${n ~/ 5000} nota(s) de R\$ 50.00");
  n = n % 5000;
  print("${n ~/ 2000} nota(s) de R\$ 20.00");
  n = n % 2000;
  print("${n ~/ 1000} nota(s) de R\$ 10.00");
  n = n % 1000;
  print("${n ~/ 500} nota(s) de R\$ 5.00");
  n = n % 500;
  print("${n ~/ 200} nota(s) de R\$ 2.00");
  n = n % 200;
  print("MOEDAS:");
  print("${n ~/ 100} moeda(s) de R\$ 1.00");
  n = n % 100;
  print("${n ~/ 50} moeda(s) de R\$ 0.50");
  n = n % 50;
  print("${n ~/ 25} moeda(s) de R\$ 0.25");
  n = n % 25;
  print("${n ~/ 10} moeda(s) de R\$ 0.10");
  n = n % 10;
  print("${n ~/ 5} moeda(s) de R\$ 0.05");
  n = n % 5;
  print("${n ~/ 1} moeda(s) de R\$ 0.01");
}
